Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm

```diff
@@ -68,25 +68,30 @@ - (instancetype)initWithResourceBaseURL:(NSURL *)resourceBaseURL
 }
 
 - (instancetype)_initWithManifestDictionary:(NSDictionary<NSString *, id> *)manifest
+{
+    return [self _initWithManifestDictionary:manifest resources:nil];
+}
+
+- (instancetype)_initWithManifestDictionary:(NSDictionary<NSString *, id> *)manifest resources:(NSDictionary<NSString *, id> *)resources
 {
     NSParameterAssert(manifest);
 
     if (!(self = [super init]))
         return nil;
 
-    API::Object::constructInWrapper<WebKit::WebExtension>(self, manifest);
+    API::Object::constructInWrapper<WebKit::WebExtension>(self, manifest, resources);
 
     return self;
 }
 
-- (instancetype)_initWithManifestData:(NSData *)manifestData
+- (instancetype)_initWithResources:(NSDictionary<NSString *, id> *)resources
 {
-    NSParameterAssert(manifestData);
+    NSParameterAssert(resources);
 
     if (!(self = [super init]))
         return nil;
 
-    API::Object::constructInWrapper<WebKit::WebExtension>(self, manifestData);
+    API::Object::constructInWrapper<WebKit::WebExtension>(self, resources);
```

Comment:
Do we need to pass a nil manifest here? Is this constructor smart enough to figure that out?

Reply:
It now looks for the manifest.JSOn entry in the resources dictionary.

Reply:
The `WebExtension::WebExtension` constructor also changed.